specification/elasticsan/ElasticSan.Management/models.tsp

```diff
@@ -0,0 +1,1051 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+using Azure.Core;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Foundations;
+
+namespace Microsoft.ElasticSan;
+
+/**
+ * The sku name.
+ */
+union SkuName {
+  string,
+
+  /**
+   * Premium locally redundant storage
+   */
+  Premium_LRS: "Premium_LRS",
+
+  /**
+   * Premium zone redundant storage
+   */
+  Premium_ZRS: "Premium_ZRS",
+}
+
+/**
+ * Provisioning state of the iSCSI Target.
+ */
+union ProvisioningStates {
+  string,
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
```

Comment:
>suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details" [](http://example.com/codeflow?start=3&length=142)

why do we need all the suppressions?

why don't we splat in the normal/required ProvisioningState states of Succeeded/Failed/Canceled, like most typespec API definitions do?